getignore.go

```diff
@@ -24,12 +25,32 @@ func (fetcher *ignoreFetcher) NameToUrl(name string) string {
 	return fetcher.baseUrl + "/" + name + ".gitignore"
 }
 
-func getNames(f *os.File, namesChannel chan string) {
+func addNamesToChannel(names []string, namesChannel chan string) {
+	for _, v := range names {
+		namesChannel <- v
+	}
+	close(namesChannel)
+}
+
+func getNamesFromArguments() []string {
+	filePointer := flag.String("file", "", "Path to file of names")
+	flag.Parse()
+	names := flag.Args()
+
+	if *filePointer != "" {
+		namesFile, _ := os.Open(*filePointer)
+		names = append(names, getNamesFromFile(namesFile)...)
```

Comment:
So I think we're doing too much in this function.

1. Creating command line flags.
2. Parsing the command line to identify the name of the file containing the ignore names.
3. Opening the file containing the ignore names.
4. Reading the content of the file and producing an array of strings.

I suggest we leave 1. (creating the CLI parser) and 2. (parsing the command line) inside a function called `parseArguments`. (I would split out 1 from 2 as well, but from the `flags` docs combining them seems to be the path of least resistance.) We return a variable called `namesFilePath` that has the path to the file containing the names.

Second, we open the `namesFile` inside of `main`, using the `namesFilePath` variable.

Third, we pass this file handle to the previously existing `getNames` function. I like the idea of renaming this function `getNamesFromFile` because that is a more descriptive name. This function will now return an array of names, since we get an array of names from `flag.Args()`. This keeps that representation consistent and it makes the channel superfluous.

Finally, we should update the signature of `NamesToUrls` to accept an array of names as input.